CodeReview

MelissaHeu opened the pull request.

Kacks2307 committed:
TESTET

Commit:
Test wieder geloescht

Commit:
niklas branch test

Commit:
niklastest geloescht

Kacks2307 committed:
hallo jo

Commit:
Branch geaddet

Commit:
Branch umbenannt

Commit:
GUI version 0.1

Commit:
GUI first version

Commit:
Merge branch 'Branch_Melissa' of https://github.com/MelissaHeu/Team05 into Branch_Niklas

Commit:
GUI version 0.2

BurakOezdemir committed:
Repo fixxen

Commit:
GUI in Framework eingebaut

BurakOezdemir committed:
Merge branch 'Branch_Niklas' of https://github.com/MelissaHeu/Team05 into Branch_Burak

RobinThatch committed:
Open properties is working

RobinThatch committed:
Prints deleted

RobinThatch committed:
method read fundamentals

BurakOezdemir committed:
Merge branch 'Branch_Robin' of https://github.com/MelissaHeu/Team05 into Branch_Burak

BurakOezdemir committed:
unfinished stuff

BurakOezdemir committed:
Inject eingefügt und code verbessert

Commit:
Added hirarchy for testing

Commit:
Added hirarchy for testing

Commit:
Test neuer Branch

Commit:
Test neuer Branch

Commit:
Merge branch 'Branch_Niklas' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

Commit:
TreeView jetzt direkt beim  Oeffnen

Commit:
TreeView jetzt direkt beim  Oeffnen

Commit:
Imports hinzugefuegt

Commit:
Imports hinzugefuegt

Commit:
FXML jetzt auch geaendert

Commit:
FXML jetzt auch geaendert

Olarto committed:
text file to treeview implemented

Olarto committed:
text file to treeview implemented

Commit:
Moved filechooser, added event listener to treeitems

Commit:
Moved filechooser, added event listener to treeitems

Commit:
Updates

Commit:
Updates

Commit:
edit function, close function, select all and unselect all implemented

Commit:
Node Klasse hinzugefügt

Commit:
Node Klasse hinzugefügt

BurakOezdemir committed:
fixing

BurakOezdemir committed:
Merge branch 'Branch_Melissa' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Burak

# Conflicts:
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/DataStructureEditorController.java
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/view/VisualizeTreeView.java

BurakOezdemir committed:
Changes from other repo merging

BurakOezdemir committed:
start of the function for the searchTextField and bug fixxing

RobinThatch committed:
better display of treeview

BurakOezdemir committed:
Eventhandler for project explorer and edited operations for Tree.

Commit:
Merge branch 'Branch_Robin' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Niklas

# Conflicts:
#	de.tu_bs.cs.isf.e4cf.compare.data_structures/META-INF/MANIFEST.MF
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/META-INF/MANIFEST.MF
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/NodeUsage.java
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/PropertiesController.java
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/ui/view/Properties.fxml

Commit:
Converted TreeView from String type to Node type

BurakOezdemir committed:
Merge branch 'master' of https://github.com/MelissaHeu/Team05 into Branch_Burak

# Conflicts:
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/META-INF/MANIFEST.MF
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/view/VisualizeTreeView.java

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

# Conflicts:
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/META-INF/MANIFEST.MF
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java

Olarto committed:
Code Cleanup and Utility Class/Stringtable

Olarto committed:
Properties Table initialized

BurakOezdemir committed:
Merge branch 'master' of https://github.com/MelissaHeu/Team05 into Branch_Burak

Commit:
PropertiesView now shows attributes and values of nodes correctly in TableView

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

# Conflicts:
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java

Commit:
trying to fix issues

Commit:
Properties table now closes when file is closed

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

# Conflicts:
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java

RobinThatch committed:
SelectAll and UnselectAll refined

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

Commit:
Ability to search tree for node added

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

# Conflicts:
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java

Commit:
Cleaned up package structure and code, encapsulated more methods in utilities class

BurakOezdemir committed:
tried to add some functions for the context menu.

BurakOezdemir committed:
Merge branch 'master' of https://github.com/MelissaHeu/Team05 into Branch_Burak

# Conflicts:
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewUtilities.java

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

merge konflikt geloest
de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java

BurakOezdemir committed:
Fixed some functionality -> Tree  can now opened in explorer

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

Commit:
reset after problem with last commit

Commit:
reset after problems with last 3 commits

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

Olarto committed:
Search function fully implemented

Olarto committed:
Merge branch 'master' of https://github.com/MelissaHeu/Team05 into Branch_Basti

Olarto committed:
new commit for search function completion

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Niklas

RobinThatch committed:
onEnter for searchField

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Niklas

RobinThatch committed:
Substring are searchable

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Niklas

RobinThatch committed:
Extended search function

Commit:
trying to fix git

BurakOezdemir committed:
fixing my branch

BurakOezdemir committed:
added delete and rename function to context menu

RobinThatch committed:
TreeViewController can´t find NodeUsage

BurakOezdemir committed:
added functions to FXML

RobinThatch committed:
Rename, delete added to TreeViewController

RobinThatch committed:
Display of Total Node Amount added

BurakOezdemir committed:
added new functions addChild and copy

Commit:
solving merge conflict

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

BurakOezdemir committed:
tried tto add functions

BurakOezdemir committed:
Merge branch 'master' of https://github.com/MelissaHeu/Team05 into Branch_Burak

# Conflicts:
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java

BurakOezdemir committed:
Added rename, addChild and copy for TreeViewController

BurakOezdemir committed:
Merge branch 'master' of https://github.com/MelissaHeu/Team05 into Branch_Burak

# Conflicts:
#	de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Niklas

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

RobinThatch committed:
rename, addChild complete - started with paste

Commit:
save added

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Niklas

BurakOezdemir committed:
Merge branch 'master' of https://github.com/MelissaHeu/Team05 into Branch_Burak

BurakOezdemir committed:
implemented save save as in menubar

Commit:
add child, add attributes implemented, rename and getInput reworked

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Niklas

Commit:
Save as reworked

Commit:
Added Cut and cleaned code

Olarto committed:
Added extract to file, multiple selections for copy/paste/cut/delete, cleaned up code, started with properties view

RobinThatch committed:
modified getTreeViewFrom tree to work recursively

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Niklas

BurakOezdemir committed:
cleaned codestrucure, fixed close bug, started to create undo, redo and edit attributes methods

RobinThatch committed:
recursive getTreeViewFromTree (static)

BurakOezdemir committed:
edited openTree and initiallizeTree

BurakOezdemir committed:
Merge branch 'master' of https://github.com/MelissaHeu/Team05 into Branch_Burak

Commit:
added helper methods for GetTreeViewFromTree

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Niklas

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/DataStructureEditorController.java:
```diff
@@ -0,0 +1,60 @@
+
+import de.tu_bs.cs.isf.e4cf.compare.data_structures_editor.stringtable.DataStructuresEditorST;
+import de.tu_bs.cs.isf.e4cf.compare.data_structures_editor.view.VisualizeTreeView;
+//import de.tu_bs.cs.isf.e4cf.compare.data_structures.editor.src.view.TreeViewController;
```
unused import

FritzJo replied:
Kommentar sollte entfernt werden

SMeatBoy replied:
Ungenutzer Import sollte entfernt werden

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		System.out.println("root: " + rootItem + "children: " + rootItem.getChildren() + tr.getRoot().getChildren());
+		treeView.setRoot(rootItem);
+//		getTreeRecursively(tr, treeView, rootItem);
+		for (Node node : tr.getLeaves()) {
```
remove comments of unused code

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		}
+
+		// tv.setRoot(rootItem);
+		treeView.setShowRoot(true);
```
remove comments of unused code

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+	
+	
+	/**
+	 * Tempor�re L�sung, noch nicht fertiggestellt
```
?

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+	}
+	
+//	public static TreeItem<NodeUsage> findTreeItemInTreeView(TreeView<NodeUsage> treeView, Node item){
+//		
```
remove comments of unused code

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/view/PropertiesView.java:
```diff
@@ -0,0 +1,5 @@
+package de.tu_bs.cs.isf.e4cf.compare.data_structures_editor.view;
+
+public class PropertiesView {
+	 
```
empty class

MelissaHeu replied:
Wird gebraucht, um die FXML zu laden

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/view/VisualizeTreeView.java:
```diff
@@ -0,0 +1,5 @@
+package de.tu_bs.cs.isf.e4cf.compare.data_structures_editor.view;
+
+public class VisualizeTreeView {
+
```
empty class

MelissaHeu replied:
Wird gebraucht, um die FXML zu laden

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/PropertiesViewUtilities.java:
```diff
@@ -0,0 +1,47 @@
+		return tableView;
+	}
+
+//	  public static void handle(MouseEvent event, TableView<Attribute> tempTable) {  
```
remove comments of unused code

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+
+	@FXML
+	void addAttribute() {
+//			treeView.getSelectionModel().getSelectedItem().getValue().addAttribute(
```
remove comments of unused code

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/PropertiesController.java:
```diff
@@ -0,0 +1,50 @@
+	}
+
+//	public void gandalf () {
+//		PropertiesViewUtilities.handle(event, propertiesTable);
```
remove comments of unused code

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/PropertiesController.java:
```diff
@@ -0,0 +1,50 @@
+//		PropertiesViewUtilities.handle(event, propertiesTable);
+//	}
+
+}
```
missing newline

FritzJo replied:
Leerzeile am Ende der Datei fehlt

SMeatBoy replied:
Newline am Dateiende fehlt

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/DataStructuresPropertiesController.java:
```diff
@@ -0,0 +1,54 @@
+
+	}
+
+}
```
missing newline

FritzJo replied:
Leerzeile am Ende der Datei fehlt

SMeatBoy replied:
Kein Newline am Dateiende

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/DataStructureEditorController.java:
```diff
@@ -0,0 +1,60 @@
+
+	}
+
+}
```
missing newline

FritzJo replied:
Leerzeile am Ende der Datei fehlt

SMeatBoy replied:
Kein Newline am Dateiende

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/NodeUsage.java:
```diff
@@ -0,0 +1,59 @@
+	public String toString() {
+		String s1 = "";
+		if (this.getAttributes().isEmpty()) {
+			return s1 + "empty";
```
just return s1 = "empty"

FritzJo replied:
s1 ist immer "", wieso die String concatenation?

SMeatBoy replied:
Vereinfachung: return "empty";

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/NodeUsage.java:
```diff
@@ -0,0 +1,59 @@
+				return attribute.getAttributeValues().iterator().next();
+			}
+			for (String s : attribute.getAttributeValues()) {
+				s1 = s;
```
deleting all other strings with this line maybe you mean s1 += s

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/NodeUsage.java:
```diff
@@ -0,0 +1,59 @@
+			return s1 + "empty";
+		}
+		for (Attribute attribute : this.getAttributes()) {
+			if(attribute.getAttributeKey().toLowerCase().equals("name")) {
```
formatting

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/NodeUsage.java:
```diff
@@ -0,0 +1,59 @@
+	}
+
+	@Override
+	public String toString() {
```
should be at the end of the file

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/PropertiesController.java:
```diff
@@ -0,0 +1,50 @@
+ */
+public class PropertiesController {
+
+	// private MouseEvent event;
```
comments of unused code

FritzJo replied:
Kommentar entfernen

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/PropertiesController.java:
```diff
@@ -0,0 +1,50 @@
+
+	/**
+	 * 
+	 * @param bool (always has to be true)
```
?

MelissaHeu replied:
wurde überarbeitet

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+				new TreeItem<NodeUsage>(new NodeUsage(tree.getRoot())));
+		treeView = TreeViewUtilities.addListener(treeView, services);
+		// totalNodeAmount
+		// .setText("Total Node Amount: " +
```
remove comments of unused code

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/NodeUsage.java:
```diff
@@ -0,0 +1,59 @@
+				s1 = s;
+			}
+		}
+		return s1;
```
Schleife scheint immer den letzten Attributwert zurückzugeben. Wieso nicht direkt das letzte Element referenzieren?

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+			currentItem = resultList.get(getSearchCounter());
+			incrementSearchCounter();
+		} else {
+			setSearchCounter(0);
+			currentItem = resultList.get(getSearchCounter());
+			incrementSearchCounter();
```
???

FritzJo replied:
getSearchCounter kann durch 0 ersetzt werden

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/PropertiesController.java:
```diff
@@ -0,0 +1,50 @@
+//	public void gandalf () {
+//		PropertiesViewUtilities.handle(event, propertiesTable);
+//	}
+
```
Kommentar entfernen

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+		treeView = TreeViewUtilities.getTreeViewFromTree(tree, this.treeView,
+				new TreeItem<NodeUsage>(new NodeUsage(tree.getRoot())));
+		treeView = TreeViewUtilities.addListener(treeView, services);
+		// totalNodeAmount
+		// .setText("Total Node Amount: " +
+		// TreeViewUtilities.searchTreeItem(treeView.getRoot(), "").size());
```
Kommentar entfernen

SMeatBoy replied:
Ungenutzer Code

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+
+	@FXML
+	void addAttribute() {
+//			treeView.getSelectionModel().getSelectedItem().getValue().addAttribute(
+//					TreeViewUtilities.getInput("Enter attribute name"),
+//					TreeViewUtilities.getInput("Enter attribute value"));
```
Kommentar entfernen

SMeatBoy replied:
Ungenutzter Code

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+ *
+ */
+public final class TreeViewUtilities {
+
```
what does this class?
seems to be using static fields as class fields

MelissaHeu replied:
utility class for the treeview

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+		treeView.refresh();
+	}
+
+	/**
+	 * 
+	 */
```
Wieso ein leerer Kommentar?

MelissaHeu replied:
Wird noch hinzugefügt

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		return currentItem;
+	}
+
+	public static int getSearchCounter() {
```
getter and setter on a  static field?

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		return searchCounter;
+	}
+
+	public static void setSearchCounter(int i) {
```
getter and setter on a  static field?

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+		}
+	}
+
+	/**
+	 * 
+	 */
```
Wieso ein leerer Kommentar?

MelissaHeu replied:
Werden noch ausgefüllt

Muehli25 commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+	 * 
+	 */
+	@FXML
+	void renameNode() { // Bug: �ndert nicht die Property sondern f�gt nur neue Property hinzu mit neuem
```
is this fixed ?

MelissaHeu replied:
ja

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+	 * 
+	 */
+	@FXML
+	void renameNode() { // Bug: �ndert nicht die Property sondern f�gt nur neue Property hinzu mit neuem
+						// Value
```
Fixed?

MelissaHeu replied:
Ja

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+	@FXML
+	void redo() {
+		System.out.println("redo");
+	}
```
Funktion/Sinn der Methoden?

MelissaHeu replied:
Werden noch implementiert

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+	public static TreeView<NodeUsage> getTreeViewFromTree(Tree tr, TreeView<NodeUsage> treeView,
+			Node item) {
+
+		recursionCounter += 1;
```
Counter wird nie zurückgesetzt, mehrfache Codeausführung führt zu Fehlern

MelissaHeu replied:
Der Counter wurde mittlerweile rausgenommen

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+			treeView.getSelectionModel().getSelectedItem().getParent().getChildren()
+					.remove(treeView.getSelectionModel().getSelectedItem());
+			displayTotalNodeAmount();
+		}
```
Funktion der Schleife nicht erkennbar? i wird innerhalb der Schleife nicht verwendet

MelissaHeu replied:
Wurde schon überarbeitet

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/build.properties:
```diff
@@ -0,0 +1,8 @@
+output.. = bin/
```
Ich glaube, ihr müsst noch die Icons und die FXML files mit exportieren

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+			displayTotalNodeAmount();
+		}
+		treeView.refresh();
+	}
+
+	/**
+	 * A method to close a file #UNFINISHED throws nullpointer when new file is
+	 * opened
+	 */
+	@FXML
+	void closeFile() {
+		// set treeview and its values to null, then remove it from the background
+		treeView.setRoot(null);
+		services.eventBroker.send("EmptyPropertiesTableEvent", true);
+	}
```
Wenn der Fehler bekannt ist, sollte er auch abgefangen werden.

MelissaHeu replied:
Haben wir schon verbessert

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/DataStructureEditorController.java:
```diff
@@ -0,0 +1,60 @@
+		canvans.setScene(scene);
+
+	}
+
```
Dead Code

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/DataStructureEditorController.java:
```diff
@@ -0,0 +1,60 @@
+	public void preDestroy() {
+
+	}
+
```
Dead Code

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/DataStructureEditorController.java:
```diff
@@ -0,0 +1,60 @@
+	public void onFocus() {
+
+	}
+
```
Dead Code

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/handler/OpenInTreeViewHandler.java:
```diff
@@ -0,0 +1,28 @@
+			services.eventBroker.send("OpenTreeEvent", reader.readArtifact(element));
+		}
+	}
+}
```
Leerzeile am Ende der Datei fehlt

SMeatBoy replied:
Kein Newline am Dateiende

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/NodeUsage.java:
```diff
@@ -0,0 +1,59 @@
+ * 
+ * @author Team05
+ * Implementation of adapter class abstract node
+ */
```
NodeUsage? Die Klasse verstehe ich nicht. Es gibt doch schon die NodeImpl warum braucht ihr die hier ?

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/PropertiesController.java:
```diff
@@ -0,0 +1,50 @@
+	 * @param bool (always has to be true)
+	 */
+	@Optional
+	@Inject
```
clearProperiesTable(...

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/PropertiesController.java:
```diff
@@ -0,0 +1,50 @@
+		propertiesTable.getColumns().clear();
+		propertiesTable = PropertiesViewUtilities.getAttributeTable(node, propertiesTable);
+	}
+
```
Oo?

MelissaHeu replied:
Haben wir vergessen zu löschen

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+
+	@FXML
+	private Label testLabel;
+
```
Zusammenfassen von Variablen gleichen Typen Text hitCount, TrotalNodeAmount;

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+	private ContextMenu contextMenu;
+
+	private List<TreeItem> copyList = new ArrayList<TreeItem>();
+
```
JavaDoc

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+import javafx.scene.control.TreeView;
+import javafx.scene.layout.VBox;
+import javafx.scene.text.Text;
+
```
JavaDoc

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+
+	@FXML
+	private TextField searchTextField;
+
```
JavaDoc Variablen sortieren nach inject und member

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+	/**
+	 * 
+	 */
+	@FXML
```
JavaDoc bei allen Methoden, die Methodennamen finde ich noch nicht so gut ich würde die mehr so addChildNode pasteNode copyNode sonst ist das etwas zu generisch.

MelissaHeu replied:
Methodennamen haben wir jetzt geändert, JavaDoc Kommentare fügen wir noch hinzu

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/PropertiesViewUtilities.java:
```diff
@@ -0,0 +1,47 @@
+		return tableView;
+	}
+
+//	  public static void handle(MouseEvent event, TableView<Attribute> tempTable) {  
+//		  if (event.getClickCount()==2) {  
+//			  
+//	    }  
+//	  }  
+}
```
Kommentar entfernen

SMeatBoy replied:
Ungenutzter Code

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		rootItem.setExpanded(true);
+		System.out.println("root: " + rootItem + "children: " + rootItem.getChildren() + tr.getRoot().getChildren());
+		treeView.setRoot(rootItem);
+//		getTreeRecursively(tr, treeView, rootItem);
```
Kommentar entfernen

SMeatBoy replied:
Unused Code

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+			rootItem.getChildren().add(item);
+		}
+
+		// tv.setRoot(rootItem);
```
Kommentar entfernen

SMeatBoy replied:
Unused Code

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+
+	/**
+	 * 
+	 * @param tr
+	 * @param treeView
+	 * @return
+	 */
```
Javadoc sollte auch Inhalt haben

MelissaHeu replied:
wurde hinzugefügt

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+
+		contextMenu.hide();
+
+		try {
```
Das hier ist etwas strenge. Warum kann da eine nullpointer exception fliegen? Kümmert euch lieber um den Grund und implementiert das error-prone

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+ * @author Team05
+ *
+ */
+public final class TreeViewUtilities {
```
Wieso ist die Class final?

MelissaHeu replied:
Die Klasse ist final weil der Ansatz so war, dass man diese Klasse nicht von außen verändern kann, weil es eine Utility Class ist.
Also so war die Idee weil nicht das jemand die dann Overrided und dadurch Methoden von der Klasse verändert obwohl diese nicht verändert werden sollten.
Hoffe es ist jetzt verständlich geworden.
Hättet ihr sonst einen besseren Ansatz?

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+//			NodeUsage x = new NodeUsage (item);
+//			if (n.getValue().getUUID().equals(x.getUUID())) {
+//				System.out.println("uuid gleich");
+//				return n;
+//			} else {
+//				continue;
+//			}
+//			
+//		}
+//		return null;
+//	}
+//	
+//	public static void treeViewToList(TreeItem<NodeUsage> item){
+//		
+//		if (!(item.getValue().getAttributes().isEmpty())){
+//			list.add(item);
+//		}
+//		
+//		if(!item.getValue().isLeaf()) {
+//			System.out.println(item + "hasChildren");
+//			for(TreeItem<NodeUsage> t : item.getChildren())
+//			treeViewToList(t);
+//		}
+//		System.out.println("treeViewToList" + list);
+//	}
+
+//	public static TreeView<NodeUsage> getTreeRecursively(Tree tr, TreeView<NodeUsage> treeView, TreeItem<NodeUsage> item) {
+//		nextItem = item;
+//		System.out.println("for schleife");
+//		//TreeItem<NodeUsage> nextItem = new TreeItem<NodeUsage>(new NodeUsage(node));
+//		System.out.println("children: " + item.getValue().getChildren());
+//		for(Node n: item.getValue().getChildren()) {
+//			nextItem.getChildren().add(new TreeItem<NodeUsage>(new NodeUsage(n)));
+//			System.out.println(n);
+//			TreeItem<NodeUsage> next = new TreeItem<NodeUsage>(new NodeUsage(n));
+//			getTreeRecursively(tr, treeView, next);
+//		}
+//	    
+//		return treeView;
+//	}
```
Kommentar entfernen

SMeatBoy replied:
Unused Code

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+
+			}
+		} else {
+//			System.out.println("DadTreeItem" + findTreeItemInTreeView(treeView, item));
+//			findTreeItemInTreeView(treeView, item).getChildren().add(new TreeItem<NodeUsage>(new NodeUsage(item)));
```
Kommentar entfernen

SMeatBoy replied:
Unused Code

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+
+		if(!item.isLeaf()) {
+			for(Node n : item.getChildren()) {
+				getTreeViewFromTree(tr, treeView, n);
```
Ergebniss der Rekursion wird verworfen! / Funktion nicht erkennbar

MelissaHeu replied:
Wurde überarbeitet

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+	
+	/**
+	 * Tempor�re L�sung, noch nicht fertiggestellt
+	 * @param tr
+	 * @param treeView
+	 * @param item
+	 * @return
```
Javadoc Inhalt fehlt

MelissaHeu replied:
wurde ergänzt

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+	}
+
+	@FXML
+	void undo() {
```
Da braucht ihr ein Stack für und ein interface da kann man sich mal das Command Pattern zu angucken ;P

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+//	}
+
+	/**
+	 * 
+	 * @param treeView
+	 * @param services
+	 * @return
+	 */
```
Javadoc leer

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		return treeView;
+	}
+
+	/**
+	 * 
+	 * @param item
+	 * @param name
+	 * @return
+	 */
```
Javadoc leer

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+	/**
+	 * search the text given in the searchfield
+	 */
+	@FXML
```
Im Allgemeinen würde ich vorschlagen, dass ihr noch ein paar inlien Kommentare macht, weil man teilweise nicht nachvollziehen kann was hier so passiert.

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+		resultList.clear();
+	}
+
+	@FXML
```
Was on Enter?

MelissaHeu replied:
Haben wir zu startSearch(...) umbenannt

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/handler/OpenInTreeViewHandler.java:
```diff
@@ -0,0 +1,28 @@
+
+public class OpenInTreeViewHandler {
+
+	/**
```
Macht da mal ein JavaDoc draus

MelissaHeu replied:
Haben wir gemacht

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		return searchList;
+	}
+
+	public static List<TreeItem<NodeUsage>> getSearchList() {
+		return searchList;
+	}
+
+	public static void clearSearchList() {
+		searchList.clear();
+	}
```
searchList ist static, wieso werden Getter verwendet?
Clear benötigt keine seperate Methode

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/handler/OpenInTreeViewHandler.java:
```diff
@@ -0,0 +1,28 @@
+	public void execute(ServiceContainer services) {
+		FileTreeElement element = services.rcpSelectionService.getCurrentSelectionFromExplorer();
+		TreeViewUtilities.treeName = services.rcpSelectionService.getCurrentSelectionFromExplorer().getRelativePath();
+		if (element.getExtension().equals("txt")) {
```
Das gefährlich und da sollte man gucken, ob es ein Reader für das Format gibt, wenn ja dann soll er das öffnen.

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/PropertiesViewUtilities.java:
```diff
@@ -0,0 +1,47 @@
+public class PropertiesViewUtilities {
+
+	/**
+	 * 
```
Fehlt die Beschreibung

MelissaHeu replied:
Haben wir hinzugefügt

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/PropertiesViewUtilities.java:
```diff
@@ -0,0 +1,47 @@
+
+		return tableView;
+	}
+
```
Dead Code

MelissaHeu replied:
Wurde entfernt

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		td.setTitle("Dialog");
+		td.showAndWait();
+		String s = td.getEditor().getText();
+		if (s.equals("") || s.equals(null)) {
+			throw new NullPointerException();
+		}
```
equals prüft auf "String Gleichheit", nicht auch Referenz

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+ *
+ */
+public final class TreeViewUtilities {
+
```
Was das hier alles? Ich glaube das gehört nicht hier hin. Teilweise sind das Parameter, die man der Rekursion mitgeben will und mehr so lokale Variablen sind. So wird dauerhaft im Speicher der Wert gehalten, obwohl das denke ich nicht nötig ist.

MelissaHeu replied:
Wurde mittlerweile entfernt

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+	public static void incrementSearchCounter() {
+		searchCounter++;
+	}
+
```
Ich glaube das kann man auch auslagern in Utilitie Klassen bzw. ich glaube das existiert auch schon

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/view/PropertiesView.java:
```diff
@@ -0,0 +1,5 @@
+package de.tu_bs.cs.isf.e4cf.compare.data_structures_editor.view;
+
```
??

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/view/PropertiesView.java:
```diff
@@ -0,0 +1,5 @@
+package de.tu_bs.cs.isf.e4cf.compare.data_structures_editor.view;
+
+public class PropertiesView {
```
??

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/view/VisualizeTreeView.java:
```diff
@@ -0,0 +1,5 @@
+package de.tu_bs.cs.isf.e4cf.compare.data_structures_editor.view;
+
```
??

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		return currentItem;
+	}
+
+	public static int getSearchCounter() {
+		return searchCounter;
+	}
+
+	public static void setSearchCounter(int i) {
+		searchCounter = i;
+	}
+
+	public static void incrementSearchCounter() {
+		searchCounter++;
+	}
```
Getter und Setter bei statischen Variablen überflüssig

SMeatBoy replied:
Direkter Zugriff auf Variable würde Lesbarkeit erhöhen

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/PropertiesController.java:
```diff
@@ -0,0 +1,50 @@
+	@FXML
+	private TableView<Attribute> propertiesTable;
+
+	/**
+	 * 
+	 * @param bool (always has to be true)
+	 */
+	@Optional
+	@Inject
+	public void emptyPropertiesTable(@UIEventTopic("EmptyPropertiesTableEvent") boolean bool) {
+		propertiesTable.getItems().clear();
+	}
```
Wieso existiert der Methodenparameter, wenn er nicht verwendet wird und auch immer True sein muss?

KamilRosiak reviewed (commented):
Also ihr müsst mal überall Java Doc Kommentare anfertigen und teilweise auch Inline Kommentare, weil es an manchen stellen nicht so verständlich ist. Lasst mal über alle Files den Eclipse Auto Formatter laufen. Sehr viel Leerzeilen und falsch eingerückte abschnitte etc.
Denkt an manchen Stellen noch über die Namen nach und der NodeUsage existiert und heißt NodeImpl.

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+	void copy() {
+		contextMenu.hide();
+		copyList.clear();
+		copyList.addAll(treeView.getSelectionModel().getSelectedItems()); // Bug: Paste ist spiegelverkehrt bei oben
```
Fixed?

MelissaHeu replied:
Ist noch nicht gefixt, aber in Planung

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		TreeItem<NodeUsage> currentItem = new TreeItem<NodeUsage>();
+		if (getSearchCounter() < resultList.size()) {
+			currentItem = resultList.get(getSearchCounter());
+			incrementSearchCounter();
```
Ein einzelnes incrementSearchCounter nach der If-Bedingung ist ausreichend.

FritzJo commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+ */
+public final class TreeViewUtilities {
+
+	static List<TreeItem<NodeUsage>> searchList = new ArrayList<TreeItem<NodeUsage>>();
+
+	public static String treeName = "";
+
+	private static int searchCounter = 0;
+
+	private static int recursionCounter = 0;
```
Verwendung von static Variablen an dieser Stelle fragwürdig

MelissaHeu replied:
Die Variablen müssen static sein, damit sie von den statischen Methoden aufgerufen werden können

Muehli25 reviewed (changes requested):
- javadoc missing or only stubs
- inconsitent commit messages (why ???)
- unused code in comments
- treeviewutils has to be reworked
- so many formatting issues
- language is not consitent (some english, some german)

This needs to be updated, before a more intense review is even possible.

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/DataStructureEditorController.java:
```diff
@@ -0,0 +1,60 @@
+
+	}
+
+	@PreDestroy
+	public void preDestroy() {
+
+	}
+
+	@Focus
+	public void onFocus() {
+
+	}
+
+	@Persist
+	public void save() {
+
+	}
```
Leere Funktionen

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/DataStructuresPropertiesController.java:
```diff
@@ -0,0 +1,54 @@
+import javafx.embed.swt.FXCanvas;
+import javafx.scene.Scene;
+
+public class DataStructuresPropertiesController {
+
+	@Inject
+	public DataStructuresPropertiesController(Composite parent, ServiceContainer services, IEclipseContext context) {
+		postConstruct(parent, services, context);
+
+	}
+
+	@PostConstruct
+	public void postConstruct(Composite parent, ServiceContainer services, IEclipseContext context) {
+		FXCanvas canvans = new FXCanvas(parent, SWT.None);
+		FXMLLoader<PropertiesView> loader = new FXMLLoader<PropertiesView>(context, DataStructuresEditorST.BUNDLE_NAME,
+				DataStructuresEditorST.PROPERTIES_VIEW_FXML);
+
+		Scene scene = new Scene(loader.getNode());
+		canvans.setScene(scene);
+
+	}
+
+	@PreDestroy
+	public void preDestroy() {
+
+	}
+
+	@Focus
+	public void onFocus() {
+
+	}
+
+	@Persist
+	public void save() {
+
+	}
```
Identisch mit DataStructuresEditorController

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/NodeUsage.java:
```diff
@@ -0,0 +1,59 @@
+			if(attribute.getAttributeKey().toLowerCase().equals("name")) {
+				return attribute.getAttributeValues().iterator().next();
+			}
+			for (String s : attribute.getAttributeValues()) {
+				s1 = s;
+			}
```
For-Loop kann durch Zugriff auf letztes Element ersetzt werden

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/PropertiesController.java:
```diff
@@ -0,0 +1,50 @@
+
+	/**
+	 * 
+	 * @param bool (always has to be true)
```
Wenn Parameter immer Wert X haben muss ist es kein Parameter?

MelissaHeu replied:
Parameter muss nicht immer true sein, der Kommentar war falsch

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/PropertiesController.java:
```diff
@@ -0,0 +1,50 @@
+		propertiesTable = PropertiesViewUtilities.getAttributeTable(node, propertiesTable);
+	}
+
+//	public void gandalf () {
+//		PropertiesViewUtilities.handle(event, propertiesTable);
+//	}
```
Ungenutzer Code

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+		treeView.refresh();
+	}
+
+	/**
+	 * 
+	 */
```
An dieser und weiteren Stellen: Leerer Kommentar

MelissaHeu replied:
JavaDoc Kommentare müssen teilweise noch ergänzt werden

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+	 * 
+	 */
+	@FXML
+	void renameNode() { // Bug: �ndert nicht die Property sondern f�gt nur neue Property hinzu mit neuem
```
Ist der Bug noch vorhanden?

MelissaHeu replied:
Nein, der ist mittlerweile resolved

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+	}
+
+	@FXML
+	void deleteNode() {
+		for (int i = 0; i < copyList.size(); i++) {
+			treeView.getSelectionModel().getSelectedItem().getParent().getChildren()
+					.remove(treeView.getSelectionModel().getSelectedItem());
+			displayTotalNodeAmount();
+		}
```
deleteNode entfernt mehrere Nodes, umbenennung in deleteNodes?
Auf i wird nicht zugegriffen

MelissaHeu replied:
Methode wurde in der Zwischenzeit bereits überarbeitet

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+	}
+
+	/**
+	 * A method to close a file #UNFINISHED throws nullpointer when new file is
```
Ist mit #UNFINISHED gemeint, dass die Implementierung nicht abgeschlossen ist?
Kommentar uneindeutig

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+	 */
+	@FXML
+	void closeFile() {
+		// set treeview and its values to null, then remove it from the background
```
Kommentar in JavaDoc der Methode verschieben

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -0,0 +1,285 @@
+		TreeViewUtilities.serializesTree(treeView, TreeViewUtilities.getInput("Save as"));
+	}
+
+	@FXML
+	void undo() {
+		System.out.println("undo");
+
+	}
+
+	@FXML
+	void redo() {
+		System.out.println("redo");
+	}
```
Umsetzung der Funktionen fehlt

MelissaHeu replied:
Das sind aktuell nur Platzhalter für die Methoden, die wir später noch implementieren.

Muehli25 replied:
Wieso habt ihr dann einen PR erstellt wenn ihr offensichtlich noch nicht fertig seid ? Speichert euch mal die Kommentare und schließt den PR und erstellt einen wenn ihr fertig seid, sonst haben wir einen erhöhten Mehraufwand wenn wir alles doppelt und dreifach reviewen müssen.

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+ * @author Team05
+ *
+ */
+public final class TreeViewUtilities {
```
Wieso final?

MelissaHeu replied:
Die Klasse ist final weil der Ansatz so war, dass  man diese Klasse nicht von außen verändern kann, weil  es eine Utility Class ist.
Also so war die Idee weil nicht das jemand die dann Overrided und dadurch Methoden von der Klasse verändert obwohl diese nicht verändert werden sollten.
Hoffe es ist jetzt verständlich geworden.
Hättet ihr sonst einen besseren Ansatz?

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+
+	static List<TreeItem<NodeUsage>> searchList = new ArrayList<TreeItem<NodeUsage>>();
+
+	public static String treeName = "";
+
+	private static int searchCounter = 0;
+
+	private static int recursionCounter = 0;
+	
+	public static List<TreeItem<NodeUsage>> list = new ArrayList<TreeItem<NodeUsage>>();
```
Nach meinem Verständnis des Codes sollten die Variablen nicht static sein

MelissaHeu replied:
Die Variablen müssen static sein, damit sie von den statischen Methoden aufgerufen werden können.

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+	public static TreeView<NodeUsage> getTreeViewFromTree(Tree tr, TreeView<NodeUsage> treeView,
+			Node item) {
+
+		recursionCounter += 1;
```
Nur erster Methodenaufruf kann zum Aufruf von Z. 77-82 führen. Ist das gewollt?

MelissaHeu replied:
Ist schon überarbeitet.

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		}
+
+		if(!item.isLeaf()) {
+			for(Node n : item.getChildren()) {
+				getTreeViewFromTree(tr, treeView, n);
+
+			}
```
Wozu Aufruf wenn return nicht genutzt wird?

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		return searchList;
+	}
+
+	public static List<TreeItem<NodeUsage>> getSearchList() {
+		return searchList;
+	}
+
+	public static void clearSearchList() {
+		searchList.clear();
+	}
```
Sinn der Funktionen unklar

MelissaHeu replied:
Wurden beide entfernt.

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+	}
+
+	public static void writeToFile(File file, TreeView treeView) {
+		if (file.getName().equals(treeName)) {
+			file.delete();
+		}
```
Eventuell Frage an User, ob Datei tatsächlich ersetzt werden soll. Anderenfalls Datenverlust möglich

MelissaHeu replied:
Wird noch implementiert.

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/view/PropertiesView.java:
```diff
@@ -0,0 +1,5 @@
+package de.tu_bs.cs.isf.e4cf.compare.data_structures_editor.view;
+
+public class PropertiesView {
+	 
+}
```
Implementierung fehlt

MelissaHeu replied:
Wird benötigt um die zugehörige FXML Datei zu laden

SMeatBoy commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/view/VisualizeTreeView.java:
```diff
@@ -0,0 +1,5 @@
+package de.tu_bs.cs.isf.e4cf.compare.data_structures_editor.view;
+
+public class VisualizeTreeView {
+
+}
```
Implementierung fehlt

MelissaHeu replied:
Wird benötigt um die zugehörige FXML Datei zu laden

SMeatBoy reviewed (changes requested):
Kommentierte Stellen sollten überarbeitet werden.
JavaDoc fehlt/unvollständig
Auskommentierter Code sollte entfernt werden
Vor allem TreeViewUtilities sollte überarbeitet werden

conqp commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/DataStructureEditorController.java:
```diff
@@ -0,0 +1,60 @@
+
+/**
+ * 
+ * @author Team05
```
Unvollständiger Javadoc.

conqp commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+				writer.write("\n");
+			}
+			writer.close();
+			System.out.println("Tree: " + file.getAbsolutePath() + " stored.");
```
Debug-printing entfernen.

MelissaHeu replied:
Wird gemacht, sobald wir die nicht mehr brauchen.

conqp commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+		td.showAndWait();
+		String s = td.getEditor().getText();
+		if (s.equals("") || s.equals(null)) {
+			throw new NullPointerException();
```
NPE unpassend für leere Zeichenkette.

MelissaHeu replied:
Wir werden noch etwas einbauen, damit dieser Fall sinnvoller behandelt wird.

conqp commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java:
```diff
@@ -0,0 +1,274 @@
+			writer.close();
+			System.out.println("Tree: " + file.getAbsolutePath() + " stored.");
+		} catch (IOException e) {
+			e.printStackTrace();
```
Stacktrace Printing als Exception Handling fragwürdig.

MelissaHeu replied:
Statt der Exception wird nun ein Fenster geöffnet, welches eine Fehlermeldung zeigt.

Commit:
commiting so i can merge

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

RobinThatch committed:
recursive getTreeViewFromTree

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

Olarto committed:
Code Review Suggestions incorporated

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

Commit:
-Added methods to change node attributes and their values
-variability class is now shown in propertiesTable
-every Node has name attribute upon creation

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Melissa

Commit:
copy works in the right order and with children

conqp commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -135,11 +141,11 @@ void copyNode() {
 		contextMenu.hide();
 		copyList.clear();
 		copyList.addAll(new ArrayList<TreeItem<NodeUsage>>(treeView.getSelectionModel().getSelectedItems())); // Bug:
```
Marked bug.

conqp commented on de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java:
```diff
@@ -135,11 +141,11 @@ void copyNode() {
-																												// bei
-																												// oben
+		System.out.println(treeView.getSelectionModel().getSelectedItem().getValue().getVariabilityClass()); // Paste
+		// ist
```
Cryptic comments.

Commit:
-attributes can now be deleted in propertiesView contextMenu
-attribute property designations can now be edited

Commit:
-contextMenu closing improved, not perfect yet

Commit:
Merge branch 'master' of https://github.com/MelissaHeu/e4CompareFramework into Branch_Niklas

Commit:
formatting

Commit:
Icons added to TreeView

RobinThatch committed:
copy repaired

Commit:
icons added

Commit:
Changed treeView and treeItems from NodeUsage to AbstractNode

Commit:
fixed issue #29 and added alert for deleting attribute
